main_test.go

```diff
@@ -0,0 +1,427 @@
+			source:      "github.com/slsa-framework/example-package",
+			pversiontag: pString("v14.1"),
+			err:         pkg.ErrorMismatchVersionedTag,
+		},
+		{
+			name:        "versioned v14.1.1 no match push-v14.2",
+			artifact:    "binary-linux-amd64-push-v14.2",
+			source:      "github.com/slsa-framework/example-package",
+			pversiontag: pString("v14.1.1"),
+			err:         pkg.ErrorMismatchVersionedTag,
+		},
+		{
+			name:        "versioned v14.3.1 no match push-v14.2",
+			artifact:    "binary-linux-amd64-push-v14.2",
+			source:      "github.com/slsa-framework/example-package",
+			pversiontag: pString("v14.3.1"),
+			err:         pkg.ErrorMismatchVersionedTag,
+		},
+		{
+			name:        "versioned v13 no match push-v14.2",
+			artifact:    "binary-linux-amd64-push-v14.2",
+			source:      "github.com/slsa-framework/example-package",
+			pversiontag: pString("v13"),
+			err:         pkg.ErrorMismatchVersionedTag,
+		},
+		{
+			name:        "versioned v15 no match push-v14.2",
+			artifact:    "binary-linux-amd64-push-v14.2",
+			source:      "github.com/slsa-framework/example-package",
+			pversiontag: pString("v15"),
+			err:         pkg.ErrorMismatchVersionedTag,
+		},
+		{
+			name:        "versioned v15.1 no match push-v14.2",
+			artifact:    "binary-linux-amd64-push-v14.2",
+			source:      "github.com/slsa-framework/example-package",
+			pversiontag: pString("v15.1"),
+			err:         pkg.ErrorMismatchVersionedTag,
+		},
+		// Special case of the e2e test repository building builder from head.
+		{
+			name:      "e2e test repository verified with builder at head",
+			artifact:  "binary-linux-amd64-e2e-builder-repo",
+			source:    "github.com/slsa-framework/example-package",
+			branch:    "main",
+			noversion: true,
+		},
+		// Malicious builders and workflows.
+		{
+			name:      "rekor upload bypassed",
+			artifact:  "binary-linux-amd64-no-tlog-upload",
+			source:    "github.com/slsa-framework/example-package",
+			err:       pkg.ErrorNoValidRekorEntries,
+			noversion: true,
+		},
+		{
+			name:      "malicious: untrusted builder",
+			artifact:  "binary-linux-amd64-untrusted-builder",
+			source:    "github.com/slsa-framework/example-package",
+			err:       pkg.ErrorUntrustedReusableWorkflow,
+			noversion: true,
+		},
+		{
+			name:      "malicious: invalid signature expired certificate",
+			artifact:  "binary-linux-amd64-expired-cert",
+			source:    "github.com/slsa-framework/example-package",
+			err:       pkg.ErrorNoValidRekorEntries,
+			noversion: true,
+		},
+		// Regression test of sharded UUID
+		{
+			name:      "regression: sharded uuids",
+			artifact:  "binary-linux-amd64-sharded",
+			source:    "github.com/slsa-framework/slsa-verifier",
+			branch:    "release/v1.0",
+			noversion: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Re-initializing variable so it is not changed while executing the closure below
+		t.Run(tt.name, func(t *testing.T) {
+			// t.Parallel()
```

Comment:
remove?

Reply:
@asraa this may be a merge problem from me. Did you intend to comment this or shall I re-enable it?

Reply:
Running the invididual tests here in parallel might result in rate-limiting, so I would avoid that

Reply:
Let me add a coment.